portal_pages/models.py

```diff
@@ -428,3 +433,103 @@ class OrganizationCaseStudy(Orderable, models.Model):
     panels = [
         FieldPanel('organization'),
     ]
+
+
+# Blog index page
+
+
+class BlogIndexPage(Page):
+    intro = RichTextField(blank=True)
+
+    search_fields = Page.search_fields + (
+        index.SearchField('intro'),
+    )
+
+    subpage_types = ['portal_pages.BlogPage']
+
+    @property
+    def blogs(self):
+        # Get list of live blog pages that are descendants of this page
+        blogs = BlogPage.objects.live().descendant_of(self)
+
+        # Order by most recent date first
+        blogs = blogs.order_by('-date')
+
+        return blogs
+
+    def get_context(self, request):
+        # Get blogs
+        blogs = self.blogs
+
+        # Filter by tag
+        tag = request.GET.get('tag')
+        if tag:
+            blogs = blogs.filter(tags__name=tag)
+
+        # Filter by author
+        author = request.GET.get('author')
+        if author:
+            blogs = blogs.filter(author__username=author)
+
+        # Pagination
+        page = request.GET.get('page')
+        paginator = Paginator(blogs, 6)  # Show 6 casestudies per page
+        try:
+            blogs = paginator.page(page)
+        except PageNotAnInteger:
+            blogs = paginator.page(1)
+        except EmptyPage:
+            blogs = paginator.page(paginator.num_pages)
+
+        # Update template context
+        context = super(BlogIndexPage, self).get_context(request)
+        context['blogs'] = blogs
+        return context
+
+BlogIndexPage.content_panels = [
+    FieldPanel('title', classname="full title"),
+    FieldPanel('intro', classname="full"),
+]
+
+BlogIndexPage.promote_panels = Page.promote_panels
+
+
+# Blog Page
+
+
+class BlogPageTag(TaggedItemBase):
+    content_object = ParentalKey('portal_pages.BlogPage', related_name='tagged_items')
+
+
+class BlogPage(Page, TopImage):
+    body = RichTextField()
+    tags = ClusterTaggableManager(through=BlogPageTag, blank=True)
+    date = models.DateField("Post date")
+
+    search_fields = Page.search_fields + (
+        index.SearchField('body'),
+    )
+
+    author = models.ForeignKey(
+        RapidProUser,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL
+    )
```

Comment:
Ultimately its wanted to have (anonymous) users submit blog entries, so I'm not sure a Foreign Key to User makes the most sense for Author field...not all blog authors will have logins on site (the only people with logins envisioned right now are the admins)...